pkg/image/admission/imagepolicy/accept.go

```diff
@@ -47,6 +48,12 @@ func accept(accepter rules.Accepter, resolver imageResolver, m meta.ImageReferen
 			// use the most generic policy rule here because we don't even know the image name
 			if attrs == nil {
 				attrs = &rules.ImagePolicyAttributes{}
+
+				// an objectref that is DockerImage ref will have a name that corresponds to its pull spec.  We can parse that
+				// to a docker image ref
+				if ref != nil && ref.Kind == "DockerImage" {
+					attrs.Name, _ = imageapi.ParseDockerImageReference(ref.Name)
```

Comment:
> why dropping the errors?

We won't return the error (resolution is optional) and if it doesn't understand it, it returns the best parse it can.